src/main/scala/org/constellation/domain/healthcheck/ReconciliationRound.scala

```diff
@@ -0,0 +1,244 @@
+            // in consequence I think we should also try to calculate cluster state from the responses we got and start consensuses for these
+            _ => (ReconciliationResult(unresponsivePeers, Set.empty, lastClusterAlignmentResult).some, ())
+          )
+
+        case Right(clusterAlignmentResults) =>
+          if (areAllNodesAligned(clusterAlignmentResults))
+            reconciliationResult.modify(
+              _ => (ReconciliationResult(Set.empty, Set.empty, clusterAlignmentResults.some).some, ())
+            )
+          else if (!isLastRun)
+            run(
+              attemptsLeft - 1,
+              if (isFirstRun) clusterAlignmentResults.some else firstClusterAlignmentResult,
+              clusterAlignmentResults.some
+            )
+          else {
+            for {
+              initiallyMisaligned <- F.delay(
+                firstClusterAlignmentResult.map(fetchMisalignedPeers).getOrElse(Set.empty)
+              )
+              currentlyMisaligned <- F.delay(fetchMisalignedPeers(clusterAlignmentResults))
+              misaligned = initiallyMisaligned.intersect(currentlyMisaligned)
+              _ <- reconciliationResult.modify(
+                _ => (ReconciliationResult(Set.empty, misaligned, clusterAlignmentResults.some).some, ())
+              )
+            } yield ()
+          }
+      }
+    } yield ()
+
+  def areAllNodesAligned(clusterAlignmentResults: Map[Id, List[ClusterAlignmentResult]]): Boolean =
+    clusterAlignmentResults.forall { case (_, result) => result == List(NodeAligned) }
+
+  def fetchMisalignedPeers(clusterAlignmentResults: Map[Id, List[ClusterAlignmentResult]]): Set[Id] =
+    clusterAlignmentResults.filter { case (id, result) => result != List(NodeAligned) }.keySet
+
+  // Should we filter out (not send) node's observation about itself?
+  private def runReconciliation(): F[Either[Set[Id], Map[Id, List[ClusterAlignmentResult]]]] =
+    for {
+      allPeers <- cluster.getPeerInfo
+      peers = allPeers.filter { case (_, peerData) => canActAsJoiningSource(peerData.peerMetadata.nodeState) }
+      // assumption and check before starting reconciliation is that there should be no rounds in progress
+      // if the round starts and reconciliation is in progress, reconciliation should be canceled and result discarded
+      ownClusterState = allPeers.mapValues(
+        pd =>
+          NodeReconciliationData(
+            pd.peerMetadata.id,
+            pd.peerMetadata.nodeState,
+            none[Set[RoundId]],
+            pd.majorityHeight.head.joined // is that correct, head???
+          )
+      )
+      responses <- peers.toList.traverse {
+        case (id, peerData) =>
+          PeerResponse
+            .run(
+              apiClient.healthcheck.getClusterState(),
+              unboundedBlocker
+            )(peerData.peerMetadata.toPeerClientMetadata.copy(port = "9003"))
```

Comment:
Take that port from config. I would avoid magic numbers.

Reply:
Yeah it would be better. I guess we should also start servers with a value from config.

Reply:
I also considered adding it to PeerData, but I remember that it may have been serialized with snapshotInfo, so without versioning it wasn't possible.